src/librustc/mir/mod.rs

```diff
@@ -664,6 +719,14 @@ impl<'tcx> BasicBlockData<'tcx> {
     pub fn terminator_mut(&mut self) -> &mut Terminator<'tcx> {
         self.terminator.as_mut().expect("invalid terminator state")
     }
+
+    pub fn retain_statements<F>(&mut self, mut f: F) where F: FnMut(&mut Statement) -> bool {
```

Comment:
Why didn't you change `erase_regions` and `clean_end_regions` to use this function?